eng/scripts/Install-LinuxArm64CrossCompileToolchain.ps1

```diff
@@ -21,29 +21,90 @@ if ($arch -ne 'x64') {
+deb [arch=arm64] http://ports.ubuntu.com/ubuntu-ports/ jammy-updates main restricted universe multiverse
+deb [arch=arm64] http://ports.ubuntu.com/ubuntu-ports/ jammy-security main restricted universe multiverse
 deb [arch=arm64] http://ports.ubuntu.com/ubuntu-ports/ jammy-backports main restricted universe multiverse
-EOF'
+EOF
+
+# Refresh package indices for both amd64 and arm64 architectures
+sudo apt-get update -qq
+
+# Find the 'gcc base package' name (there is only one 'gcc base package' name per Ubuntu series across all architectures)
+GCC_BASE=$(apt-cache search --names-only '^gcc-[0-9]+-base$' | awk '{print $1}' | sort -V | tail -1)
+
+if [[ -z "$GCC_BASE" ]]; then
+  echo "ERROR: Could not determine gcc base package." >&2
+  exit 1
+fi
+
+# Find 'gcc base package' candidate versions in both architectures
+amd64_list=$(apt-cache madison "${GCC_BASE}:amd64" | awk '{print $3}')
+arm64_list=$(apt-cache madison "${GCC_BASE}:arm64" | awk '{print $3}')
+
+amd64_gcc_versions=$(printf "%s\n" "$amd64_list" | sort -V)
+arm64_gcc_versions=$(printf "%s\n" "$arm64_list" | sort -V)
 
-sudo sed -i -e 's/deb http/deb [arch=amd64] http/g' /etc/apt/sources.list
-sudo sed -i -e 's/deb mirror/deb [arch=amd64] mirror/g' /etc/apt/sources.list
+# Use the highest common version of 'gcc base package'
+gcc_version=$(
+  comm -12 \
+    <(printf "%s\n" "$amd64_gcc_versions") \
+    <(printf "%s\n" "$arm64_gcc_versions") \
+  | tail -1
+)
 
-sudo apt update
-sudo apt install -y clang llvm binutils-aarch64-linux-gnu gcc-aarch64-linux-gnu zlib1g-dev:arm64
-"@
+if [[ -z "$gcc_version" ]]; then
+  echo "ERROR: No common ${GCC_BASE} version across amd64 and arm64." >&2
+  echo "amd64 candidates:" >&2; printf "%s\n" "$amd64_gcc_versions" >&2
+  echo "arm64 candidates:" >&2; printf "%s\n" "$arm64_gcc_versions" >&2
+  exit 1
+fi
+
+echo "Using ${GCC_BASE} version: ${gcc_version} (selected from amd64 and arm64 candidates)"
+
+# Install gcc base libraries for both amd64 and arm64 with versions pinned
+# Note: 'libgcc-s1' and 'gcc-*-base' are 'Multi-Arch:same', which means they must be installed at the exact same version across all enabled architectures
+sudo DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+  ${GCC_BASE}:amd64=${gcc_version} \
+  ${GCC_BASE}:arm64=${gcc_version} \
+  libgcc-s1:amd64=${gcc_version} \
+  libgcc-s1:arm64=${gcc_version}
+
+# Install libc6 and rest of toolchain
+# Since we already installed and pinned 'libgcc-s1:arm64' to match the 'libgcc-s1:amd64' version, version alignment is guaranteed and 'libc6:arm64' can now be installed safely.
+sudo DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+  libc6:arm64 \
+  clang \
+  llvm \
+  lld \
+  binutils-aarch64-linux-gnu \
+  gcc-aarch64-linux-gnu \
+  zlib1g-dev:arm64
+
+# Verification step
+dpkg -l | grep -E '^(ii)\s+(libc6|libgcc-s1|gcc-[0-9]+-base):arm64' || true
```

Comment:
`dpkg -l | grep <package_name>`

could you add comment document the regex for the next person please

Reply:
done